src/rp2_common/hardware_clocks/clocks.c

```diff
@@ -123,6 +123,26 @@ bool clock_configure(clock_handle_t clock, uint32_t src, uint32_t auxsrc, uint32
     return true;
 }
 
+bool clock_configure_mhz(clock_handle_t clock, uint32_t src, uint32_t auxsrc, uint16_t src_freq_mhz, uint16_t freq_mhz) {
+    assert(src_freq_mhz >= freq_mhz);
+
+    if (freq_mhz > src_freq_mhz)
+        return false;
+
+    uint32_t div = (uint32_t)((((uint32_t) src_freq_mhz) << CLOCKS_CLK_GPOUT0_DIV_INT_LSB) / freq_mhz);
+#if PICO_RP2040
+    // on RP2040 only clock divider of 1, or  >= 2 are supported
+    if (div < (2u << CLOCKS_CLK_GPOUT0_DIV_INT_LSB)) {
+        div = (1u << CLOCKS_CLK_GPOUT0_DIV_INT_LSB);
+    }
+#endif
+    uint32_t actual_freq = (uint32_t) ((((uint32_t) src_freq_mhz) << CLOCKS_CLK_GPOUT0_DIV_INT_LSB) / div) * MHZ;
+
+    clock_configure_internal(clock, src, auxsrc, actual_freq, div);
+    // Store the configured frequency
```

Comment:
What does this comment mean? (and is it referring to the line below or the line above the comment?)

Reply:
I’m not entirely sure (it was copied from the function above) - it’s related to the line above it as that’s what stores the configured frequency